feat(plex): add latest IBM Plex package for updates

seejamescode opened the pull request:
## Overview

Resolves https://github.ibm.com/Bluemix/carbon-issues/issues/553

Adds `@ibm/plex` as a dependency to receive latest updates as soon as they are available. I did keep all old Plex files since you host them on a CDN for Carbon users.

One question I do have for the maintainers: Do you also want font face declarations for the italic files? If a Carbon user does have italics, they currently will not use the right Plex italic files.

### Added

- `@ibm/plex`
- Gulp task to copy all IBM Plex Mono and Sans files to the src

### Changed

- Assumes the user always wants Plex even if they also want Helvetica. This is due to not being able to do conditional imports in scss. I hope this is okay since Plex is transitioning to default anyways.
- Font face declarations are established inside of `@ibm/plex` instead of Carbon. We specify which font face declarations we want.


## Testing / Reviewing

To see successful build:
- `yarn install`
- `yarn run build`
- Review content of compiled CSS

I have trouble seeing the result locally because the prefix of fonts is to Carbon’s CDN.

seejamescode committed:
feat(plex): add latest IBM Plex package for updates

joshblack commented:
Was hoping to use the `scss` files coming from IBM/Plex instead of bringing in the fonts themselves. Is that a supported use-case with the switch?

asudoh commented on src/globals/scss/_css--plex-core.scss:
```diff
@@ -1,52 +1,9 @@
-$font-path: 'https://unpkg.com/carbon-components@latest/src/globals/fonts' !default;
+$font-prefix: 'https://unpkg.com/carbon-components@latest/src/globals/fonts' !default;
```
Do we want to have Carbon keep shipping Plex fonts? Or do we want to change this font path ref here as well as one in `_css—font-face.scss` to choose the right font path for Plex and the legacy font? CC @joshblack

asudoh replied:
Oh Josh seems to have addressed this question already 😉

asudoh reviewed (commented):
Seems that the dev env (with `gulp serve`) is broken, to determine how to fix that, we need our team’s input wrt shipping Plex font.

seejamescode commented:
> Was hoping to use the scss files coming from IBM/Plex instead of bringing in the fonts themselves. Is that a supported use-case with the switch?

I did import IBM/Plex scss files. Do you mean we should use unpkg for the Plex package? That definitely makes sense.

asudoh commented:
Thank you for asking @seejamescode! While @joshblack may have more detailed answer, here's my perspective; AFAIK `carbon-components` has Plex fonts in its `src` directory as of today just because there was no good way to directly use an external package for Plex font (@carbon-design-system/ui please correct me if I'm wrong here). Therefore if Plex is distributed in NPM, which seems definitely the case looking at this PR, we can just stop putting Plex fonts in the `src` directory.

The `unpkg` reference works as the default font path, and the font path for demo stuffs. I'd expect every consumer overrides it for their production code. So I'm guessing referring to Plex in `unpkg` in the variables when Plex font should be in use would make sense. Thanks!

seejamescode committed:
fix(fonts): remove importing of font files from plex package

seejamescode committed:
feat(fonts): added italic references for IBM Plex

seejamescode committed:
fix(fonts): updated font-family rules

seejamescode committed:
fix(demo): import plex font face declarations in demo instead of mixin

seejamescode committed:
fix(fonts): version the carbon font file path

seejamescode commented:
Alright y’all, got some updates for you:

- Reference `unpkg` for the Plex package. I do version it instead of `latest` in case any breaking changes happen.
- I am scared to remove the existing Plex files in Carbon because users on older Carbon versions are pinging the `latest` tag. I have fixed the reference in this PR.
- Added italic references, but Anna did want to make sure @marijohannessen is okay with that

marijohannessen commented:
I'm okay with that - I thought not including italics was a restriction from our designers so as long as they are okay with it then I am too 👍

asudoh reviewed (commented):
Looks great - Thank you @seejamescode for making the change! One last comment I have is wrt Sass `@import` - We’d want to set `includePaths` in our gulp-sass setting so that we don’t need to use `../../../node_modules` path. Thanks!

seejamescode committed:
chore(fonts): add includePaths for node_modules

seejamescode commented:
Done!

asudoh reviewed (approved):
💯 Thanks @seejamescode!! 👍

tw15egan reviewed (approved):
Looks great! Thanks for adding that in, @seejamescode! 👍 ✅

marijohannessen reviewed (approved):
Looks great! 👍 ✅ 🎉
